core/common/src/Clock.kt

```diff
@@ -32,12 +58,25 @@ public interface Clock {
 
 /**
  * Returns the current date at the given [time zone][timeZone], according to [this Clock][this].
+ *
+ * The time zone is important because the current date is not the same in all time zones at the same time.
```

Comment:
Since the example shows the use of an `Instant`, maybe you can use instant instead of time.

> the current date is not the same in all time zones at the same instant